diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt

```diff
@@ -61,6 +75,439 @@ class ASTNodeUtilsTest {
+            }
+        }
+    }
+
+    @Test
+    fun `test hasChildOfType`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getIdentifierName() != null)
+                Assert.assertTrue(node.hasChildOfType(IDENTIFIER))
+        }
+    }
+
+    @Test
+    fun `test hasAnyChildOfTypes`(){
+        val code = """
+            class Test {
+                val x = 0
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.getAllChildrenWithType(IDENTIFIER).isNotEmpty() || node.getAllChildrenWithType(CLASS).isNotEmpty()) {
+                val param = arrayOf(IDENTIFIER, CLASS)
+                Assert.assertTrue(node.hasAnyChildOfTypes(*param))
+            }
+        }
+    }
+
+    @Test
+    fun `test findChildBefore`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.findChildBefore(CLASS_BODY, CLASS) != null)
+                Assert.assertEquals(node.findChildBefore(CLASS_BODY, CLASS)!!.text, code)
+        }
+    }
+
+    @Test
+    fun `test findChildAfter`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if (node.findChildAfter(VALUE_PARAMETER_LIST, TYPE_REFERENCE) != null)
+                Assert.assertEquals(node.findChildAfter(VALUE_PARAMETER_LIST, TYPE_REFERENCE)!!.text, "Int")
+        }
+    }
+
+    @Test
+    fun `test allSiblings`(){
+        val code = """
+            class Test() {
+                /**
+                * test method
+                * @param a - dummy int
+                */
+                fun foo(a: Int): Int = 2 * a
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            val list = mutableListOf<IElementType>()
+            node.allSiblings(true).forEach {
+                list.add(it.elementType)
+            }
+            Assert.assertTrue(list.contains(node.elementType))
+        }
+    }
+
+    @Test
+    fun `test isNodeFromCompanionObject`() {
+        var code = """
+            object DataProviderManager {
+                fun registerDataProvider(provider: DataProvider) {}
+
+                val allDataProviders: Collection<DataProvider>
+            }
+        """.trimIndent()
+        applyToCode(code){node ->
+            if(node.elementType == FUN)
+                Assert.assertTrue(node.isNodeFromCompanionObject())
```

Comment:
This method doesn't actually check if node is from companion object, it checks if it is from any object. Please ask Andrey if this method should be corrected or renamed. If it should be corrected, it should look for `companion` keyword in modifier list of the object.

Reply:
yes, this is bug. https://github.com/cqfn/diKTat/issues/70